kubernetes/helm_charts/secor/templates/secor-statefulset.yaml

```diff
@@ -0,0 +1,124 @@
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ .Release.Name }}
+  namespace: {{ .Values.secor_namespace }}
+  labels:
+    app: secor
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  serviceName: secor
+  updateStrategy:
+    type: RollingUpdate
+  replicas: {{ dig $.Release.Name "replicas" "" $.Values.secor_jobs }}
+  selector:
+    matchLabels:
+      app: secor
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: secor
+        release: {{ .Release.Name }}
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "9102"
+        prometheus.io/path: "/metrics"
+    spec:
+      containers:
+        - name: {{ .Release.Name }}-secor
+          image: {{ .Values.image.prefix | default "" }}{{ .Values.image.repository }}:{{ .Values.image.tag }}
+          # image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: "CONFIG_FILE"
+              value: "{{ .Values.secor.partition.configFile }}"
+            - name: "LOG4J_CONFIGURATION"
+              value: "{{ .Values.secor.partition.logConfigFile }}"
+            - name: "JVM_MEMORY"
+              value: "{{ .Values.secor.partition.jvm.memory }}"
+
+          volumeMounts:
+           - mountPath: /opt/secor/config
+             name: secor-config
+           - mountPath: /opt/secor/config/logs
+             name: secor-log-config
+           - name: "data-path"
+             mountPath: /mnt/{{ dig $.Release.Name "service_name" "" $.Values.secor_jobs }}/message_logs/partition
+          resources:
+{{ toYaml .Values.secor.partition.resources | indent 12 }}
+
+      volumes:
+       - configMap:
+           name: {{ .Release.Name }}-config
+         name: secor-config
+       - configMap:
+           name: secor-log-config
+         name: secor-log-config
+
+  volumeClaimTemplates:
+    - metadata:
+        name: data-path
+        labels:
+          app: secor
+        annotations:
+          volume.beta.kubernetes.io/storage-class: "{{ .Values.secor.partition.storage.class }}"
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: "{{ .Values.secor.partition.storage.size }}"
+
+---
+apiVersion: apps/v1
```

Comment:
can we split this a separte filelike `statsd-deployment.yml` instead of clubbing them?

Reply:
I will move it to a separate deployment manifest file.